PublicMapping/districtbuilder
# Select geometries at any geolevel

## Overview
Allow for selecting geometries at any geolevel (counties + tracts/blockgroups at the same time, or just blocks) and add a tooltip so it's clear to the user why certain geolevels are deactivated.

### Demo
![db_geolevel_editing](https://user-images.githubusercontent.com/2926237/89814562-4febc580-db11-11ea-8ca3-3da6940bc58a.gif)

![db_subgeounit_selection](https://user-images.githubusercontent.com/2926237/90064013-2a4cf080-dcb8-11ea-8b2e-878ab1eef1c9.gif)

### Notes
This includes two different issues since they were related and #267 turned out to basically be a one-line change to get rid of auto-accepting since everything about selecting features at different geolevels as part of the same selection already worked (EDIT: except what commenters pointed out). #268 included some rules about not being able to select both the base geolevel (blocks) and some other geolevel at the same time as well as informative tooltips with messaging to explain what was going on to the user.

After asking @jfrankl about the tooltip, we decided to move away from showing which criteria were met vs not met and new mocks were created for the tooltip (see https://github.com/PublicMapping/districtbuilder/issues/268#issuecomment-671511801) 

These mocks don't cover the acceptance criteria in #268 numbers 4 and 5 which describe the tooltip for the non-zoom level limited geolevels, but I kept it consistent in terms of the `<conditions> to edit <geolevel>` pattern.

This also includes two very small unrelated changes (cb838ed and f9cb79e)

## Testing Instructions
* While zoomed out, select a blockgroup/tract and a county as part of the same selection
* Ensure the tooltip text for "Blocks" tells you to zoom in and resolve the changes
* Accept the changes and make sure that works fine
* Now the 'Blocks' tooltip text should tell you just to zoom in
* Select a county/blockgroup/tract
* Now the 'Blocks' tooltip text should tell you just to resolve changes
* Cancel the selection
* Zoom to the block level and now you should be able to select 'Blocks'
* Select a block and mouse over the disabled buttons and you should see the tooltip tell you to zoom out to edit those geolevels
* Accept or cancel and all geolevels should be enabled
* Select an intermediate geolevel geounit (blockgroups/tracts)
* Switch to 'County' and select the whole county and ensure that the county selection envelops the smaller geounit selection correctly (need to test with both selection tools)
* Select an entire county and then switch to intermediate geolevel and ensure geounits can be selected/deselected correctly and that this saves correctly
* Check that demographic totals are being summed up correctly when a selection includes geounits from multiple geolevels

Closes #267 
Closes #268
